network/p2p/p2p.go

```diff
@@ -115,9 +115,14 @@ func MakeHost(cfg config.Local, datadir string, pstore *pstore.PeerStore) (host.
 		listenAddr = "/ip4/0.0.0.0/tcp/0"
 	}
 
-	// the libp2p.NoListenAddrs builtin disables relays but this one does not
-	var noListenAddrs = func(cfg *libp2p.Config) error {
-		cfg.ListenAddrs = []multiaddr.Multiaddr{}
+	// the libp2p.NoListenAddrs builtin disables relays but this one only does if
```

Comment:
this is technically right but I think two separate opts `noListenAddrNoRelay := libp2p.NoListenAddrs(p2pCfg)` and `noListenAddrs := func... p2pCfg.ListenAddrs = []multiaddr.Multiaddr{}` could be more clear?